lib/src/models/scoped_class_declaration.dart

```diff
@@ -0,0 +1,28 @@
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:meta/meta.dart';
+
+import 'class_type.dart';
+
+/// Represents a declaration of a class / mixin / extension.
+@immutable
+class ScopedClassDeclaration {
+  /// Type of declared class entity.
+  final ClassType type;
+
+  /// The node that represents in the AST structure for a Dart program.
```

Comment:
Seems that there is something missing after `represents` or I didn't catch the meaning...